javascript/javascript1/week4/index.js

```diff
@@ -0,0 +1,145 @@
+    answer2:"you can search on:https://www.hackyourfuture.dk/",
+    answer3:"enghavevej 80 2450 københavn sv",
+    answer4:"fishing added to your todo",
+    answer5:"Removed fishing from your todo",
+    answer6:"fishing and singing in the shower",
+    answer7:()=>{
+        const date= new Date();
+        const year = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(date);
+        const month = new Intl.DateTimeFormat('en', { month: 'long' }).format(date);
+        const day = new Intl.DateTimeFormat('en', { day: '2-digit' }).format(date);
+        const currentDate = `${day}. Of ${month} ${year}`;
+         return currentDate;
+    },
+    answer8:(str)=>{
+        const patternForNum=/\d+/g;
+        const patternForOperator=/([\+\*\-\/])/;   
+        const  di = str.match(patternForNum);
+        const  op = str.match(patternForOperator);
+        const a=parseInt(di[0]);
+        const b=parseInt(di[1]);
+        const operator=op[0];
+    switch (operator) {
+        case '+':
+            return a + b;
+        case '-':
+            return a - b;
+        case '*':
+            return a * b;
+        case '/':
+           return a / b;
+         }
+    },
+
+    answer9:(str)=>{
+          const pattern=/\d+/;
+           let time=parseInt(str.match(pattern));
+         setInterval(y=() => {
+             return "Done"                 
+         }, time*1000);
+         return y();
+    } 
+}
+
+function replace(re) // function use  to remove spaces.
+{
+    if(re)
+    {
+     return  re.replace(/\s+/g,'')
+    }
+
+}
+
+
+function getReply(command)// getReply function body
+{
+    
+    if(replace(questions.question1).localeCompare(replace(command))==0)
+    {
+       
+         return answers.answer1(command);
+         
+    }
+    if(replace(questions.question2).localeCompare(replace(command))==0)
+    {
+         return answers.answer2;
+    }
+    if(replace(questions.question3).localeCompare(replace(command))==0)
+    {
+         return answers.answer3;
+    }
+    if(replace(questions.question4).localeCompare(replace(command))==0)
+    {
+         return answers.answer4;
+
+    }
+    if(replace(questions.question5).localeCompare(replace(command))==0)
+    {
+         return answers.answer5;
+    }
+    if(replace(questions.question6).localeCompare(replace(command))==0)
+    {
+         return answers.answer6;
+    }
+    if(replace(questions.question7).localeCompare(replace(command))==0)
+    {
+         return answers.answer7();
+    }
+    if(replace(questions.question8).localeCompare(replace(command))==0)
+    {
+        return answers.answer8(command);
+        
+    }
+    if(replace(questions.question9).localeCompare(replace(command))==0)
+    {
+        
+        return (answers.answer9(command))
+    }
+    else{
+
+        return "OOps!I can't recognize you please follow the  manual of this software"
+    }    
+}
+
+const reply=getReply("Hello my name is Benjamin")// calling of getReply function
```

Comment:
You have hardcoded all questions which is something we always should try and avoid in programming. Fx if i say Hello my name is Peter the program would fail which is not what it should do. It should automatically find the name and use that.